### DISCUSSION TITLE
UxPlay 1.62

### DISCUSSION THREAD

**Discussion by fduncanh** [FDH2/UxPlay#172] [Announcements]
UxPlay is a GPLv3 unix AirPlay2 Mirror and AirPlay2 Audio (but not AirPlay2 Video) server: its main use is to act like an AppleTV for screen-mirroring (with audio) of iOS/MacOS clients (iPads, iPhones, MacBooks, etc.) in a window on the server display (with the possibility of sharing that window on screen-sharing applications such as Zoom) on a host running Linux, macOS, or other unix, (and now also Windows 10/11, with the unix-like MYSYS2 environment) using Apple's AirPlay2 Mirror protocol. When the client screen is not mirrored, Apple Lossless (ALAC) AirPlay2 Audio can be streamed from the client to the server.

UxPlay is derived from https://github.com/FD-/RPiPlay, with audio and video rendered using GStreamer, and various enhancements. While originally targeted at X86/X86_64 server and desktop systems, it now also supports the GStreamer Video4Linux2 plugin for 32- and 64-bit accelerated hardware video decoding on Raspberry Pi (replacing 32-bit-only no-longer-maintained OpenMAX used by RPiPlay). UxPlay is tested on a number of systems, including (among others) Debian 10.11 "Buster" and 11.2 "Bullseye", Ubuntu 20.04 LTS and 22.04.1 LTS, Linux Mint 20.3, Pop!_OS 22.04 (NVIDIA edition), Rocky Linux 8.6 (a CentOS successor), Fedora 36, OpenSUSE 15.4, Arch Linux 22.10, macOS 12.3 (Intel and M1), FreeBSD 13.1. On Raspberry Pi, it is tested on Raspberry Pi OS (Bullseye) (32- and 64-bit), Ubuntu 22.10, and Manjaro RPi4 22.12. Also tested on 64-bit Windows 10 and 11.

1.62 2023-01-22 Added Audio-only mode time offset -ao x to allow user synchronization in Audio-only mode of audio playing on the server with video, song lyrics, etc. playing on the client. x = 5.0 appears to be optimal in many cases. Quality fixes: cleanup in volume changes, timestamps, some bugfixes.

<hr /><em>This discussion was created from the release <a href='https://github.com/FDH2/UxPlay/releases/tag/v1.62'>UxPlay 1.62</a>.</em>

**Comment by IInatas** [comment]
Hi 
I compiled uxplay on Manjaro via AUR package - no errors so far.
`sudo uxplay` give the following messages in a terminal:

> UxPlay 1.62: An Open-Source AirPlay mirroring and audio-streaming server.
> using system MAC address 81:a5:28:19:d4:82
> Initialized server socket(s)
> error: XDG_RUNTIME_DIR is invalid or not set in the environment.
> Accepted IPv4 client on socket 32
> Local: 192.168.178.37
> Remote: 192.168.178.39
> Client identified as User-Agent: AirPlay/665.13.1
> Accepted IPv4 client on socket 34
> Local: 192.168.178.37
> Remote: 192.168.178.39
> raop_rtp_mirror starting mirroring
> raop_rtp_mirror tcp socket is closed, got 0 bytes of 128 byte header
> raop_rtp_mirror->running is no longer true
> Connection closed for socket 32
> Connection closed for socket 34
> error: XDG_RUNTIME_DIR is invalid or not set in the environment.
> ^CStopping...

Connecting an IPAD Air works, but no window on the laptop is opened.
I tried  `sudo uxplay -vs ximagesink ` also without success.

What option do I have to get it working again?


regards,
Marcus

**Reply by fduncanh** [reply]
DONT USE "sudo uxplay", just "uxplay"

(I think that's your problem)
Only use sudo when you need to (for installing)

"XDG_RUNTIME_DIR is invalid or not set in the environment" is because the  graphical environment belongs to you, the user, not "root"  (the sudo user)

ps, 
please post in issues, not the "discussion" (you can reply here, this time, if you want to)

**Reply by IInatas** [reply]
I am adding a comment here to make it easier for others to find in case of the following problem  (see more on issue #173).
`uxplay`
>UxPlay 1.62: An Open-Source AirPlay mirroring and audio-streaming server.
Required gstreamer plugin 'libav' not found
Missing plugin is contained in '[GStreamer 1.x]-libav'
stopping

A workaround that works for me (on manjaro).
rm -rf ~/.cache/gstreamer-1.0/ 

hth
